### PR TITLE
Add util.Settings to load configuration from file and update samples

### PR DESCRIPTION
**Pull request by milderhc**
- Add Settings class to load configuration from file.
- Update samples.
- Rename samples following dotnet style.

### CODE REVIEW

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/util/Settings.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        try (FileInputStream fis = new FileInputStream(Settings.getAbsolutePath())) {
+            Properties props = new Properties();
+            props.load(fis);
+            if (defaultValue == null) {
+                return props.getProperty(propertyName);
+            }
+            return props.getProperty(propertyName, defaultValue);
+        } catch (IOException e) {
+            LOGGER.error(
+                    "Unable to load config value " + propertyName + " from file " + SettingsFile,
```
Comment (dsgrieve) [inline java/semantickernel-api/src/main/java/com/microsoft/semantickernel/util/Settings.java]: If you log the settingsFile, you leak details about the environment that this is being run in, which is a possible attack vector. Something to consider.

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/util/Settings.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                Settings.getSettingsValue(path, Property.AZURE_OPEN_AI_ENDPOINT.label),
+                Settings.getSettingsValue(path, Property.AZURE_OPEN_AI_DEPLOYMENT_NAME.label, ""));
+    }
+
+    private static String getSettingsValue(String SettingsFile, String propertyName) {
+        return getSettingsValue(SettingsFile, propertyName, null);
+    }
+
+    private static String getSettingsValue(
+            String SettingsFile, String propertyName, String defaultValue) {
```
Comment (dsgrieve) [inline java/semantickernel-api/src/main/java/com/microsoft/semantickernel/util/Settings.java]: s/SettingsFile/settingsFile/ to be more consistent with Java conventions

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/util/Settings.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    /**
+     * Returns an instance of OpenAISettings with key and organizationId from the properties file
+     *
+     * @param path Path to the properties file
+     * @return OpenAISettings
+     */
+    public static OpenAISettings getOpenAISettingsFromFile(String path) {
+        return new OpenAISettings(
+                Settings.getSettingsValue(path, Property.OPEN_AI_KEY.label),
```
Comment (dsgrieve) [inline java/semantickernel-api/src/main/java/com/microsoft/semantickernel/util/Settings.java:79]: Not that it matters so much at the moment, but can we just load the properties file once?

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/util/Settings.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            props.load(fis);
+            if (defaultValue == null) {
+                return props.getProperty(propertyName);
+            }
+            return props.getProperty(propertyName, defaultValue);
+        } catch (IOException e) {
+            LOGGER.error(
+                    "Unable to load config value " + propertyName + " from file " + SettingsFile,
+                    e);
+            throw new RuntimeException(SettingsFile + " not configured properly");
```
Comment (dsgrieve) [inline java/semantickernel-api/src/main/java/com/microsoft/semantickernel/util/Settings.java]: throw the IOException. I'd throw the IOException all the way back to the caller of getAzureOpenAISettingsFromFile

**Comment by johnoliver** [comment]
I will probably merge this (with some review fixes) however we are soon to merge upstream (probably today), so dont want to take any additions at the moment, and will either apply your changes for you when we are merged, or get you to resubmit on the upstream project

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/util/Settings.java (hunk 2 above)
Comment (milderhc) [reply java/semantickernel-api/src/main/java/com/microsoft/semantickernel/util/Settings.java]: Replace mistake

**Pull request closed without merging**